File: java/src/test/java/com/example/livecoding/twopointers/ReverseStringTest.java
package com.example.livecoding.twopointers;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;

class ReverseStringTest {

    @Test
    void happy() {
        char[] chars = "hello".toCharArray();
        ReverseString.reverseString(chars);
        assertArrayEquals("olleh".toCharArray(), chars);
    }

    @Test
    void single() {
        char[] chars = {'a'};
        ReverseString.reverseString(chars);
        assertArrayEquals(new char[]{'a'}, chars);
    }

    @Test
    void empty() {
        char[] chars = {};
        ReverseString.reverseString(chars);
        assertArrayEquals(new char[]{}, chars);
    }

    @Test
    void evenLength() {
        char[] chars = "abcd".toCharArray();
        ReverseString.reverseString(chars);
        assertArrayEquals("dcba".toCharArray(), chars);
    }
}

File: typescript/tests/reverseString.test.ts
import { describe, expect, it } from "vitest";

import { reverseString } from "../src/two_pointers/reverseString.js";

describe("reverseString", () => {
  it("happy", () => {
    const chars = "hello".split("");
    reverseString(chars);
    expect(chars.join("")).toBe("olleh");
  });

  it("single", () => {
    const chars = ["a"];
    reverseString(chars);
    expect(chars).toEqual(["a"]);
  });

  it("empty", () => {
    const chars: string[] = [];
    reverseString(chars);
    expect(chars).toEqual([]);
  });

  it("even length", () => {
    const chars = "abcd".split("");
    reverseString(chars);
    expect(chars.join("")).toBe("dcba");
  });
});
